packages/synapse-react-client/src/components/ColumnFilter/ColumnFilter.test.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import { QueryResultBundle } from '@sage-bionetworks/synapse-types'
+import { fireEvent, render, screen, waitFor } from '@testing-library/react'
+import ColumnFilter from './ColumnFilter'
+import syn20337467Json from '../../../src/mocks/query/syn20337467.json'
+import { useQuery } from '@tanstack/react-query'
+import { useQueryContext } from '../QueryContext/QueryContext'
+
+jest.mock('../QueryContext/QueryContext', () => ({
+  useQueryContext: jest.fn(),
+}))
+jest.mock('@tanstack/react-query', () => ({
+  useQuery: jest.fn(),
+  QueryClient: jest.fn().mockImplementation(() => ({
+    queryCache: { clear: jest.fn() },
+    setQueryData: jest.fn(),
+    getQueryData: jest.fn(),
+  })),
+}))
+
```

Comment:
I like how you approached making this a unit test. That said, we've tended to avoid mocking the entire react-query module because it becomes difficult to create mocks in components that contain multiple instances of `useQuery`. If we were to extend this component with another query, we'd have to totally change this mock approach, building dynamic logic into `useQuery` mock.

Instead, I think you could continue to extend the QueryContext mock by making the queryMetadataQueryOptions something like this:

```ts
{
  queryKey: ['queryMetadataQueryOptions']
  queryFn: () => data
}
```

It looks like we've actually done this in another test, FullTextSearch.test.ts

Reply:
We don't typically do it, but I like how you mocked `useQueryContext` instead of using the real one, so no need to change that.

Reply:
When you use the real react-query, you'll need to wrap the component under test in a QueryContext. We created a context wrapper for all tests, so when you call `render`,  you can just do

```tsx
render(<ColumnFilter {...props} />, { wrapper: createWrapper() })
```

`createWrapper` will come from TestingLibraryUtils.tsx